Library/Homebrew/formulary.rb

```diff
@@ -237,18 +245,74 @@ def self.loader_for(ref)
     return NullLoader.new(ref)
   end
 
+  def self.find_with_priority(ref, behavior)
+    available_formulas = tap_paths(ref)
+    core_formula = core_path(ref)
+    if core_formula.file?
+      available_formulas[Tap.core_priority] = [] if available_formulas[Tap.core_priority].nil?
+      available_formulas[Tap.core_priority] << core_formula
+    end
+    unless available_formulas.empty?
+      if behavior == FactoryBehavior::ENFORCE_UNIQUE
+        flat_formulas = available_formulas.values.flatten
+        if flat_formulas.length == 1
+          return flat_formulas.first
+        else
+          raise TapFormulaAmbiguityError.new(ref, flat_formulas)
+        end
+      else
+        if available_formulas.length > 1
+          opoo "Some formulae are excluded due to priority settings. Run brew info #{ref} to see them."
+        end
+        available_formulas.keys.sort.each do |this_priority|
+          if available_formulas[this_priority].length > 1
+            if behavior == FactoryBehavior::INSTALL_LIKE
+              ohai "Multiple formulae with same name and priority available."
+              formulae_list = available_formulas[this_priority]
+              puts_columns (0..formulae_list.length-1).map { |i| "#{i+1}. #{formulae_list[i].fully_qualified_name}"}
+              print "Please choose one by inputing the number before it: "
+              selected_index = $stdin.readline.to_i - 1
```

Comment:
Discussed with @asparagui just now. It seems that we have to decide a way to make a selection between formulae within same priority if we don't want interactive shell when installing.

So far the best way I can think of is to select following the order the taps get tapped. (More precisely, the time the priority index is set.)

In that case, we can warn the user when he or she tries to set a 'conflicting' priority for a tap, saying something like 'Tap(s) with the same priority already exist: Formulae in this tap might get shadowed. You may wish to change its priority by running xxx.'

Any idea on this? I would like to hear more suggestion about it :)